CRM/MembershipExtras/Hook/Pre/Contribution.php

```diff
@@ -54,19 +54,29 @@ public function preProcess() {
    * Checks if total amount is ok vs sum of line items.
    */
   private function rectifyAmountsBasedOnLineItems() {
-    $lineItems = $this->getContributionLineItems();
-    $totalAmount = 0;
-    $taxAmount = 0;
+    $givenTotalAmount = CRM_Utils_Array::value('total_amount', $this->params, NULL);
+    $givenTaxAmount = CRM_Utils_Array::value('tax_amount', $this->params, NULL);
+
+    /*
+     * If amount is not being changed WE SHOULD NOT UPDATE AMOUNT! Doing so will
+     * cause line items to be updated, using total amount of contribution as
+     * line total and adding taxes again!
+     */
+    if (!isset($givenTotalAmount)) {
+      return;
+    }
 
+    $calculatedTotalAmount = 0;
+    $calculatedTaxAmount = 0;
+    $lineItems = $this->getContributionLineItems();
     foreach ($lineItems as $line) {
       $lineTax = CRM_Utils_Array::value('tax_amount', $line, 0);
-      $totalAmount += $line['line_total'] + $lineTax;
-      $taxAmount += $lineTax;
+      $calculatedTotalAmount += $line['line_total'] + $lineTax;
+      $calculatedTaxAmount += $lineTax;
     }
-
-    if ($totalAmount != $this->params['total_amount'] || $taxAmount != $this->params['tax_amount']) {
-      $this->params['total_amount'] = $totalAmount;
-      $this->params['tax_amount'] = $taxAmount;
+    if ($calculatedTotalAmount != $givenTotalAmount || $calculatedTaxAmount != $givenTaxAmount) {
```

Comment:
I don't think this condition is relevant and can be removed , if these amout does indeed match then assigning them to the params won't affect anything,